lib/taskana-core/src/test/java/acceptance/task/QueryTaskWithAttachment.java

```diff
@@ -0,0 +1,99 @@
+package acceptance.task;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+
+import java.util.List;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import acceptance.AbstractAccTest;
+import pro.taskana.Attachment;
+import pro.taskana.AttachmentSummary;
+import pro.taskana.Task;
+import pro.taskana.TaskService;
+import pro.taskana.TaskSummary;
+import pro.taskana.exceptions.InvalidArgumentException;
+import pro.taskana.exceptions.NotAuthorizedException;
+import pro.taskana.exceptions.TaskNotFoundException;
+import pro.taskana.security.JAASRunner;
+import pro.taskana.security.WithAccessId;
+
+/**
+ * Acceptance test for the usecase of adding/removing an attachment of a task and update the result correctly.
+ */
+@RunWith(JAASRunner.class)
+public class QueryTaskWithAttachment extends AbstractAccTest {
+
+    public QueryTaskWithAttachment() {
+        super();
+    }
+
+    @WithAccessId(
+            userName = "user_1_1",
+            groupNames = {"group_1"})
+    @Test
+    public void testGetAttachmentSummariesFromTask() {
+        TaskService taskService = taskanaEngine.getTaskService();
+        List<TaskSummary> tasks = taskService.createTaskQuery()
+                .classificationKeyIn("L110102")
+                .list();
+        assertEquals(1, tasks.size());
+
+        List<AttachmentSummary> attachmentSummaries = tasks.get(0).getAttachmentSummaries();
+        assertNotNull(attachmentSummaries);
+        assertEquals(2, attachmentSummaries.size());
+    }
+
+    @WithAccessId(
+            userName = "user_1_2")
+    @Test
+    public void testGetNoAttachmentSummaryFromTask() {
+        TaskService taskService = taskanaEngine.getTaskService();
+        List<TaskSummary> tasks = taskService.createTaskQuery()
+                .list();
+        assertEquals(20, tasks.size());
+
+        List<AttachmentSummary> attachmentSummaries = tasks.get(0).getAttachmentSummaries();
+        assertEquals(null, attachmentSummaries);
```

Comment:
this should not be the case. please, if there are no values in a list, return an empty list and not null